src/hooks/useApproveConfirmTransaction.tsx

```diff
@@ -7,7 +7,7 @@ import useCatchTxError from './useCatchTxError'
 type LoadingState = 'idle' | 'loading' | 'success' | 'fail'
 
 type Action =
-  | { type: 'requires_approval' }
+  | { type: 'has_approval' }
```

Comment:
Although it works fine, this action is hard to be understood why requires_approval makes approvalState success. So I changed the name and condition

Reply:
It seems like `has_approval` and `approve_receipt` are the same. Could we consolidate them ?

Reply:
Done